modules/infra/control/port.c

```diff
@@ -309,6 +309,28 @@ int iface_port_reconfig(
 			return errno_log(-ret, "rte_eth_macaddr_get");
 	}
 
+	if ((set_attrs & GR_PORT_ADD_MAC) && !eth_addr_is_zero(&api->mac)) {
+		struct rte_ether_addr mac;
+		memcpy(&mac, &api->mac, sizeof(mac));
+		if (rte_is_multicast_ether_addr(&mac))
+			ret = gr_port_mcast_addr_add(p, &mac);
+		else
+			ret = gr_port_unicast_addr_add(p, &mac);
+		if (ret)
+			return errno_log(-ret, "add mac address");
+	}
+
+	if ((set_attrs & GR_PORT_REM_MAC) && !eth_addr_is_zero(&api->mac)) {
```

Comment:
If the flag is set, should you actually check the value of the mac address? The other functions will fail if it is invalid.